app/controllers/user_status_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class UserStatusController < ApplicationController
+  requires_login
+
+  def set
+    ensure_feature_enabled
+    raise Discourse::InvalidParameters.new(:description) if params[:description].blank?
+
+    current_user.set_status(params[:description])
+    render json: success_json
+  end
+
+  def clear
+    ensure_feature_enabled
+    current_user.clear_status
+    render json: success_json
+  end
+
+  private
+
+  def ensure_feature_enabled
+    raise ActionController::MethodNotAllowed.new if !SiteSetting.enable_user_status
```

Comment:
I'm curious what the reason is for choosing `MethodNotAllowed` which returns a `405`. I kind of think `403` is the more correct response here when the site setting is disabled since the method is actually allowed but the server is choosing not to fulfill the request not because of the request method.

Reply:
Yes 403 would work for me too, I was mostly annoyed by 404

Reply:
No strong dislike :) These semantics are always a little bit hard to follow, I would vote for consistency we core, if we usually go with 403 for this, go with 403, if 404 go for 404...

Reply:
Searching through core, it seems like we return a 404 as a convention.

Reply:
Ok, we go with 404.